controllers/goharbor/registry/deployments.go

```diff
@@ -0,0 +1,319 @@
+package registry
+
+import (
+	"context"
+	"path"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	goharborv1alpha2 "github.com/goharbor/harbor-operator/apis/goharbor.io/v1alpha2"
+)
+
+const (
+	apiPort                          = 5000 // https://github.com/docker/distribution/blob/749f6afb4572201e3c37325d0ffedb6f32be8950/contrib/compose/docker-compose.yml#L15
+	metricsPort                      = 5001 // https://github.com/docker/distribution/blob/b12bd4004afc203f1cbd2072317c8fda30b89710/cmd/registry/config-dev.yml#L34
+	VolumeName                       = "registry-config"
+	ConfigPath                       = "/etc/registry"
+	CompatibilitySchema1Path         = ConfigPath + "/compatibility-schema1"
+	CompatibilitySchema1VolumeName   = "compatibility-schema1-certificate"
+	AuthenticationHTPasswdPath       = ConfigPath + "/auth"
+	AuthenticationHTPasswdVolumeName = "authentication-htpasswd"
+	StorageName                      = "storage"
+	StoragePath                      = "/var/lib/registry"
+)
+
+var (
+	varFalse = false
+	varTrue  = true
+)
+
+func (r *Reconciler) GetDeployment(ctx context.Context, registry *goharborv1alpha2.Registry) (*appsv1.Deployment, error) { // nolint:funlen
+	image, err := r.GetImage(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot get image")
+	}
+
+	name := r.NormalizeName(ctx, registry.GetName())
+	namespace := registry.GetNamespace()
+
+	envs := []corev1.EnvVar{}
+
+	volumes := []corev1.Volume{
+		{
+			Name: VolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: name,
+					},
+					Optional: &varFalse,
+				},
+			},
+		},
+	}
+
+	volumesMount := []corev1.VolumeMount{
+		{
+			Name:      VolumeName,
+			MountPath: ConfigPath,
+		},
+	}
+
+	if registry.Spec.HTTP.SecretRef != "" {
```

Comment:
Fix #61

Reply:
Glad to see the issue can be fixed here. But I'm thinking we may need to cherry-pick this fix and the other ones you mentioned below to the 0.5 branch and release a new patch version. Let's have a talk later.